internal/cephfs/snapshot.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2020 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cephfs
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ceph/ceph-csi/internal/util"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	klog "k8s.io/klog/v2"
+)
+
+// cephfsSnapshot represents a CSI snapshot and its cluster information.
+type cephfsSnapshot struct {
+	NamePrefix string
+	Monitors   string
+	// MetadataPool & Pool feilds are not used atm. But its definitely good to have it in this struct
+	// so keeping it here
+	MetadataPool string
+	Pool         string
+	ClusterID    string
+	RequestName  string
+}
+
+func createSnapshot(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, snapID, volID volumeID) error {
```

Comment:
what is volID and why is it not part of the volOptions?

Reply:
VolID is the subvolume. VolOptions is the metadata information of cluster + FS + Group info.